src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,41 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> { // minor comment
+
+    private static final int MIN_ELIGIBLE_AGE = 35;
+    private static final String REQUIRED_NATIONALITY = "Ukrainian";
+    private static final int MIN_YEARS_IN_UKRAINE = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate == null) {
```

Comment:
let's make it simplier
```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(REQUIRED_NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```

create method 
```
private boolean checkTimeLivingInCountry(Candidate candidate) {
        String[] dates = candidate.getPeriodsInUkr().split(SEPARATOR);
        return Integer.parseInt(dates[END_INDEX])
                - Integer.parseInt(dates[BEGIN_INDEX]) >= PERIOD_IN_UKR;
    }
```